src/Agent.Worker/Release/ReleaseDirectoryManager.cs

```diff
@@ -22,31 +30,57 @@ public sealed class ReleaseDirectoryManager : AgentService, IReleaseDirectoryMan
             ArgUtil.NotNull(projectId, nameof(projectId));
             ArgUtil.NotNull(releaseDefinition, nameof(releaseDefinition));
 
-            ReleaseDefinitionToFolderMap map;
-            string mapFile = Path.Combine(
+            ReleaseTrackingConfig trackingConfig;
+            string trackingConfigFile = Path.Combine(
                 workingDirectory,
                 Constants.Release.Path.RootMappingDirectory,
                 collectionId,
                 projectId,
                 releaseDefinition,
                 Constants.Release.Path.DefinitionMapping);
 
-            Trace.Verbose($"Mappings file: {mapFile}");
-            map = LoadIfExists(mapFile);
-            if (map == null)
+            Trace.Verbose($"Mappings file: {trackingConfigFile}");
+            trackingConfig = LoadIfExists(trackingConfigFile);
+            if (trackingConfig == null)
             {
                 Trace.Verbose("Mappings file does not exist. A new mapping file will be created");
                 var releaseDirectorySuffix = ComputeFolderInteger(workingDirectory);
-                map = new ReleaseDefinitionToFolderMap();
-                map.ReleaseDirectory = string.Format(
+                trackingConfig = new ReleaseTrackingConfig();
+                trackingConfig.ReleaseDirectory = string.Format(
                     "{0}{1}",
                     Constants.Release.Path.ReleaseDirectoryPrefix,
                     releaseDirectorySuffix);
-                WriteToFile(mapFile, map);
-                Trace.Verbose($"Created a new mapping file: {mapFile}");
+                trackingConfig.UpdateJobRunProperties();
+                WriteToFile(trackingConfigFile, trackingConfig);
+                Trace.Verbose($"Created a new mapping file: {trackingConfigFile}");
+            }
+
+            return trackingConfig;
+        }
+
+        public async Task RunMaintenanceOperation(IExecutionContext executionContext)
```

Comment:
does this needs to be `async`?

Reply:
Since the usage of the method was having await in it, had to add async for avoiding build failures